ingest/src/main/scala/hydra.ingest/http/TopicDeletionEndpoint.scala

```diff
@@ -0,0 +1,146 @@
+package hydra.ingest.http
+
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.directives.Credentials
+import akka.http.scaladsl.server.{ExceptionHandler, Route}
+import cats.data.{NonEmptyList, Validated, ValidatedNel}
+import hydra.ingest.programs.{DeleteTopicError, KafkaDeletionErrors, SchemaDeletionErrors, TopicDeletionProgram}
+import hydra.common.util.Futurable
+import hydra.core.http.RouteSupport
+import spray.json.{DefaultJsonProtocol, RootJsonFormat}
+import hydra.core.monitor.HydraMetrics.addPromHttpMetric
+import hydra.ingest.programs.TopicDeletionProgram.SchemaDeleteTopicErrorList
+
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+final class TopicDeletionEndpoint[F[_]: Futurable] (deletionProgram: TopicDeletionProgram[F], deletionPassword: String)
+  extends RouteSupport with
+    DefaultJsonProtocol with
+    SprayJsonSupport {
+
+  implicit val endpointFormat = jsonFormat2(DeletionEndpointResponse.apply)
+  case class DeletionEndpointResponse(topicOrSubject: String, message: String)
+
+  private def tupleErrorsToResponse(errorTuple: (List[SchemaDeletionErrors], List[KafkaDeletionErrors])): List[DeletionEndpointResponse] = {
+    schemaErrorsToResponse(errorTuple._1) ::: kafkaErrorsToResponse(errorTuple._2)
+  }
+
+  private def schemaErrorsToResponse(schemaResults: List[SchemaDeletionErrors]): List[DeletionEndpointResponse] = {
+    schemaResults.flatMap(_.schemaDeleteTopicErrorList.errors.map(e => DeletionEndpointResponse(e.getSubject, e.errorMessage)).toList)
+  }
+
+  private def kafkaErrorsToResponse(kafkaResults: List[KafkaDeletionErrors]): List[DeletionEndpointResponse] = {
+    kafkaResults.flatMap(_.kafkaDeleteTopicErrorList.errors.map(e => DeletionEndpointResponse(e.topicName, e.errorMessage)).toList)
+  }
+
+  final case class DeletionRequest(topics: List[String])
+  implicit val deleteRequestFormat: RootJsonFormat[DeletionRequest] = jsonFormat1(DeletionRequest)
+
+  private def validResponse(topics: List[String], userDeleting: String)(implicit ec: ExecutionContext) = {
+    topics.map(topicName => addPromHttpMetric(topicName, StatusCodes.OK.toString(), "/deleteTopics/"))
+    topics.map(topicName => log.info(s"User $userDeleting deleted topic: $topicName"))
```

Comment:
This is just an FYI, I don't know if there is any advantage to doing it this way.. But with log.x, you can also interpolate using the syntax:

`log.info("User {} deleted topic: {}", userDeleting, topicName)`

Also, you should use foreach here and for the prom metric since you are not using the result of the map (log.info returns unit). And I guess it would probably be better to do both of these in a single foreach to save on iterations (though there won't ever be that many topics so it may not matter haha)